grails-app/domain/org/pih/warehouse/invoice/InvoiceItem.groovy

```diff
@@ -82,8 +82,17 @@ class InvoiceItem implements Serializable {
         product(nullable: true)
         glAccount(nullable: true)
         budgetCode(nullable: true)
-
-        quantity(nullable: false, min: 0) // min = 0 for canceled items
+        quantity(nullable: false, min: 0, validator: { quantity, obj ->
+            InvoiceItem.withNewSession {
+                InvoiceItem notUpdatedInvoiceItem = InvoiceItem.get(obj?.id)
```

Comment:
This scares me a little bit. 

I've only ever seen code that does a findBy or exists. 

https://github.com/mcroteau/greenfield-ecommerce/blob/974442fc4dbe2c7a79a8c825313d53f5cef23791/grails-app/domain/org/greenfield/AccountRole.groovy#L91

For now, make this object read-only so we're not doing anything untoward. 
```
InvoiceItem.read(obj.id)
```
My concern is that while we're creating a new Hibernate session we're still participating in the same transaction and I don't know how things will behave if something unforeseen happens.

If this does cause any issues while testing, perhaps we need a different way to get the original value 
* grails should have a property where it stores the original values (preferred)
* named query on InvoiceItem to return just the value we want i.e. original invoiceItem.quantity (hate it, but would be better than hydrating the full object)

Reply:
I found what I was looking for ... getPersistentValue https://docs.grails.org/latest/ref/Domain%20Classes/getPersistentValue.html 

https://stackoverflow.com/questions/5005932/grails-custom-validator-based-on-a-previous-value-of-the-field

Reply:
Let's use that if we can.